server/channels/app/imaging/utils.go

```diff
@@ -140,8 +142,156 @@ func FillImageTransparency(img image.Image, c color.Color) {
 	}
 }
 
+// CropAnchor cuts out a rectangular region with the specified size
+// from the image using the specified anchor point and returns the cropped image.
+// Adapted from github.com/disintegration/imaging
+func CropCenter(img image.Image, w, h int) image.Image {
+	srcBounds := img.Bounds()
+	anchorPoint := image.Pt(srcBounds.Min.X+(srcBounds.Dx()-w)/2, srcBounds.Min.Y+(srcBounds.Dy()-h)/2)
+	r := image.Rect(0, 0, w, h).Add(anchorPoint)
+	b := srcBounds.Intersect(r)
+	return transform.Crop(img, b)
+}
+
+// cropAndResize crops the image to the smallest possible size that has the required aspect ratio using
+// a centered anchor point, then scales it to the specified dimensions and returns the transformed image.
+//
+// This is generally faster than resizing first, but may result in inaccuracies when used on small source images.
+// Adapted from github.com/disintegration/imaging
+func cropAndResizeCenter(img image.Image, width, height int) image.Image {
+	dstW, dstH := width, height
+
+	srcBounds := img.Bounds()
+	srcW := srcBounds.Dx()
+	srcH := srcBounds.Dy()
+	srcAspectRatio := float64(srcW) / float64(srcH)
+	dstAspectRatio := float64(dstW) / float64(dstH)
+
+	var tmp image.Image
+	if srcAspectRatio < dstAspectRatio {
+		cropH := float64(srcW) * float64(dstH) / float64(dstW)
+		tmp = CropCenter(img, srcW, int(math.Max(1, cropH)+0.5))
+	} else {
+		cropW := float64(srcH) * float64(dstW) / float64(dstH)
+		tmp = CropCenter(img, int(math.Max(1, cropW)+0.5), srcH)
+	}
```

Comment:
Here, we are adding 0.5 after doing the math.Max, but in Resize, we are doing it inside math.Max. Can you clarify?